ng add @angular/pwa generate an unhandled exception occurred: Cannot find module '@schematics/angular/utility'

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Oh hi there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->

# 🐞 Bug report

### Command
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [x] generate
- [x] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description
Command "ng add @angular/pwa" generate an unhandled exception occurred: Cannot find module '@schematics/angular/utility'

## 🔬 Minimal Reproduction

<!--
Simple steps to reproduce this bug.

Please include: commands run (including args), packages added, related code changes.

If reproduction steps are not enough for reproduction of your issue, please create a minimal GitHub repository with the reproduction of the issue.
A good way to make a minimal reproduction is to create a new app via `ng new repro-app` and add the minimum possible code to show the problem.
Share the link to the repo below along with step-by-step instructions to reproduce the problem, as well as expected and actual behavior.

Issues that don't have enough info and can't be reproduced will be closed.

You can read more about issue submission guidelines here: https://github.com/angular/angular-cli/blob/main/CONTRIBUTING.md#-submitting-an-issue
-->

ng add @angular/pwa --project my.project

## 🔥 Exception or Error
ℹ Using package manager: npm
✔ Found compatible package version: @angular/pwa@14.0.0.
✔ Package information loaded.

The package @angular/pwa@14.0.0 will be installed and executed.
Would you like to proceed? Yes
✔ Package successfully installed.
An unhandled exception occurred: Cannot find module '@schematics/angular/utility'
Require stack:
- myproject/node_modules/@angular/cli/models/schematic-engine-host.js
See "/private/var/folders/kk/7f9l6gdn0mjcvzmzbn9vfxz40000gn/T/ng-lcv9gi/angular-errors.log" for further details.
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 13.3.7
Node: 16.14.0
Package Manager: npm 8.3.1
OS: darwin arm64

Angular: 13.3.11
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
... service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1303.7
@angular-devkit/build-angular   13.3.7
@angular-devkit/core            13.3.7
@angular-devkit/schematics      13.3.7
@angular/cli                    13.3.7
@schematics/angular             13.3.7
rxjs                            7.5.5
typescript                      4.6.4
</code></pre>

--- a/packages/angular/pwa/package.json
+++ b/packages/angular/pwa/package.json
@@ -15,5 +15,13 @@
     "@angular-devkit/schematics": "0.0.0-PLACEHOLDER",
     "@schematics/angular": "0.0.0-PLACEHOLDER",
     "parse5-html-rewriting-stream": "6.0.1"
+  },
+  "peerDependencies": {
+    "@angular/cli": "^13.0.0"
+  },
+  "peerDependenciesMeta": {
+    "@angular/cli": {
+      "optional": true
+    }
   }
 }

--- a/tests/legacy-cli/e2e/tests/build/material.ts
+++ b/tests/legacy-cli/e2e/tests/build/material.ts
@@ -1,5 +1,5 @@
 import { getGlobalVariable } from '../../utils/env';
-import { replaceInFile } from '../../utils/fs';
+import { readFile, replaceInFile } from '../../utils/fs';
 import { installPackage, installWorkspacePackages } from '../../utils/packages';
 import { ng } from '../../utils/process';
 import { isPrereleaseCli, updateJsonFile } from '../../utils/project';
@@ -7,7 +7,7 @@ import { isPrereleaseCli, updateJsonFile } from '../../utils/project';
 const snapshots = require('../../ng-snapshot/package.json');
 
 export default async function () {
-  const tag = await isPrereleaseCli() ?  '@next' : '';
+  let tag = (await isPrereleaseCli()) ? '@next' : '';
   await ng('add', `@angular/material${tag}`, '--skip-confirmation');
 
   const isSnapshotBuild = getGlobalVariable('argv')['ng-snapshots'];
@@ -25,10 +25,15 @@ export default async function () {
       dependencies['@angular/material-moment-adapter'] =
         snapshots.dependencies['@angular/material-moment-adapter'];
     });
-
     await installWorkspacePackages();
   } else {
-    await installPackage('@angular/material-moment-adapter');
+    if (!tag) {
+      const installedMaterialVersion = JSON.parse(await readFile('package.json'))['dependencies'][
+        '@angular/material'
+      ];
+      tag = `@${installedMaterialVersion}`;
+    }
+    await installPackage(`@angular/material-moment-adapter${tag}`);
   }
 
   await installPackage('moment');

